### ISSUE TITLE
Error 256 running command

### ISSUE THREAD

**Issue by zhaoxvwahaha** [tseemann/shovill#59]
there is an error when I running the shovill, anybody could gimme a help?
Read stats: max_len = 100
Read stats: avg_len = 100
Read stats: total_bp = 1200311600
Read stats: min_len = 100
Estimating genome size with 'kmc'
Running: kmc -m32 -n256 -ci3 -k25 -t16 mypair1.gz kmc /tmp/9619922.1.st.q >> 20-kmc.log 2>&1
**Error 256 running command**

**Comment by tseemann** [comment]
It seems `kmc` is not working on your system.

Can you paste the contents of the `20-kmc.log` file here please?

**Comment by zhaoxvwahaha** [comment]
I run several some samples, but all report the 256 error, like this:
Checking for assembly errors in scaffolds.fasta
Running: bwa index scaffolds.fasta >> 70-bwa.log 2>&1
Running: (bwa mem -v 3 -x intractg -t 16 scaffolds.fasta R1.fq.gz R2.fq.gz | samtools sort --threads 1 -m 100G --reference scaffolds.fasta -T /tmp/9838024.1.st.q/samtools.58882 -o shovill.bam) >> 70-bwa.log 2>&1
Error 256 running command

The command I used is:
shovill --outdir abc --R1 pair1.fq.gz --R2 pair2.fq.gz --ram 200 --minlen 100 --asm scaffolds --trimopt LEADING:20 TRAILING:20

**Comment by zhaoxvwahaha** [comment]
I run several other samples, but all report the 256 error, like this:
Checking for assembly errors in scaffolds.fasta
Running: bwa index scaffolds.fasta >> 70-bwa.log 2>&1
Running: (bwa mem -v 3 -x intractg -t 16 scaffolds.fasta R1.fq.gz R2.fq.gz | samtools sort --threads 1 -m 100G --reference scaffolds.fasta -T /tmp/9838024.1.st.q/samtools.58882 -o shovill.bam) >> 70-bwa.log 2>&1
Error 256 running command

The command I used is:
shovill --outdir abc --R1 pair1.fq.gz --R2 pair2.fq.gz --ram 200 --minlen 100 --asm scaffolds --trimopt LEADING:20 TRAILING:20

**Comment by tseemann** [comment]
This is now a different error?
I notice samtools is asking for `-m 100G` of RAM ?
Do you have 100 gigabytes of fre RAM on your computer?

**Comment by zhaoxvwahaha** [comment]
Thank you for timely reply,
All my tasks have the same error as I posted above.
Because I submit the job by SGE to super computer cluster, perhaps the RAM is available when the source is vacant. The error 256 is  related to the RAM problem? How to set the RAM of samtools, I don't think it need so large RAM, this value is automatically compute by your pipeline?
The complete original command line is: software/anaconda2/bin/shovill --outdir abc --R1 pair1.fq.gz --R2 pair2.fq.gz --ram 200 --minlen 100 --asm scaffolds --trimopt LEADING:20 TRAILING:20

**Comment by tseemann** [comment]
Please do **not** use `--ram 200` **unless** your computer nodes have 200 GB of RAM.  

I am unable to help you with setting the correct parameters for your SGE queue.

**Comment by giriarteS** [comment]
I got the same error running kmc
`Hello xxx`
`You ran: /usr/local/bin/shovill --outdir shovel --R1 Clav04_S1_L001_R1_001.fastq.gz --R2 Clav04_S1_L001_R2_001.fastq.gz`
`This is shovil 0.9.0`
`Written by Torsten Seemann <torsten.seemann@gmail.com>`
`Homepage is https://github.com/tseemann/shovill`
`Operating system is darwin`
`Using seqtk - /usr/local/bin/seqtk`
`Using pigz - /usr/local/bin/pigz`
`Using kmc - /usr/local/bin/kmc`
`Using trimmomatic - /usr/local/bin/trimmomatic`
`Using lighter - /usr/local/bin/lighter`
`Using flash - /usr/local/bin/flash`
`Using spades.py - /usr/local/bin/spades.py`
`Using bwa - /usr/local/bin/bwa`
`Using samtools - /usr/local/bin/samtools`
`Using java - /usr/bin/java`
`Using pilon - /usr/local/bin/pilon`
`Using tee - /usr/bin/tee`
`Using sysctl - /usr/sbin/sysctl`
`Machine has 24 CPU cores and 64 GB RAM`
`Changing into folder: /Users/xxx/Desktop/Clav04/shovill`
`Collecting raw read statistics with 'seqtk'`
`Running: seqtk fqchk -q3 \/Users\/BrodersLab\/Desktop\/Clav04\/Clav04_S1_L001_R1_001\.fastq\.gz > 10-seqtk.tab`
`Read stats: total_bp = 433406240`
`Read stats: max_len = 151`
`Read stats: min_len = 151`
`Read stats: avg_len = 151`
`Estimating genome size with 'kmc'`
`Running: kmc -m8 -sm -n256 -ci3 -k25 -t16 \/Users\/xxx\/Desktop\/Clav04\/Clav04_S1_L001_R1_001\.fastq\.gz kmc /var/folders/mp/rjltg4g177v1jqwywpw2gjn00000gn/T/PJj2WG5Olm >> 20-kmc.log 2>&1`
`Error 256 running command`

Here you can see the `20-kmc.log` file:

`*********************************`
`Error: Cannot open temporary file /var/folders/mp/rjltg4g177v1jqwywpw2gjn00000gn/T/PJj2WG5Olm/kmc_00253.bin`

**Comment by tseemann** [comment]
@giriarteS what version of kmc are you using?
```
% kmc
K-Mer Counter (KMC) ver. 3.0.0 (2017-01-28)
```

**Comment by tseemann** [comment]
@stephenturner can you send me the logs?  did you use the `--ram` option ?

**Comment by giriarteS** [comment]
@tseeman,
I am using shovill 0.9 and kmc Version 3.0.0 (2017-01-28).

**Comment by MohamedRefaat92** [comment]
I'm having the same problem. I downloaded shovill yesterday and it worked fine on different samples. Today, when I executed the same command on the same data, I get this error after java :
```
Running: _JAVA_OPTIONS=-Xmx8g pilon --genome contigs.fasta --frags shovill.bam --fix bases --output pilon --threads 1 --changes --mindepth 0.25 >> 80-pilon.log 2>&1
Error 256 running command
```
This is the command I used.
`shovill --outdir mrsa --R1 1_S1_L001_R1_001.fastq.gz  --R2 1_S1_L001_R2_001.fastq.gz 
`
I have to say that I tried to fix the problem by uninstalling then reinstalling shovill and it's dependencies, but I still get the same error.

**Comment by tseemann** [comment]
@giriarteS that's good, but i would like to see the actual error message in `shovel/20-kmc.log` file? 
do you still have it?

**Comment by tseemann** [comment]
@stephenturner your error is `Message 'Cannot open temporary file to write'.` 
in `shovill --help` what is your `--tmpdir` set to?  and does it have any space?
This folder gets passed to spades.   You also have some wierd linker errors with conda spades?
Does it fail on the binaries you download from the Spades site?

**Comment by tseemann** [comment]
@MohamedRefaat92 Can you show me what the following command say?
* `pilon --version`
* `which pilon`
* `head -n3 $(which pilon)`
* `java -version`

**Comment by MohamedRefaat92** [comment]
`$ pilon --version`
`Pilon version 1.22 Wed Mar 15 16:38:30 2017 -0400
`

`$which pilon`
`/home/linuxbrew/.linuxbrew/bin/pilon
`

`$head -n3 $(which pilon)
`

```
#!/bin/bash
exec java -Xmx1000m -Xms20m -jar /home/linuxbrew/.linuxbrew/Cellar/pilon/1.22/pilon-1.22.jar "$@"

```

`$ java -version`

```
openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-8u162-b12-0ubuntu0.16.04.2-b12)
OpenJDK 64-Bit Server VM (build 25.162-b12, mixed mode)
```

**Comment by tseemann** [comment]
@MohamedRefaat92 that all looks the same as mine. I am working on a new version of Shovill right now and will release in 6 hours or so.  It will have much better output to help find the bug.

**Comment by giriarteS** [comment]
Below you can see the error message in shovel/20-kmc.log file:

Error: Cannot open temporary file /var/folders/mp/rjltg4g177v1jqwywpw2gjn00000gn/T/PJj2WG5Olm/kmc_00253.bin

I tried to run shovill again addeding the genome size option. I got a different error (a segmentation fault) with lighter.

**Comment by tseemann** [comment]
@giriarteS  
Are you using a very old Mac computer? Or a very new one?
How are you installing all the software?  via Homebrew?
```
Operating system is darwin
Machine has 24 CPU cores and 64 GB RAM
```

**Comment by giriarteS** [comment]
I used homebrew to install shovill in a new Mac Pro.

**Comment by stephenturner** [comment]
My issue was the tmpdir. I defaulted to using /tmp, which only has a few gb
available. Changing that solved my problem.

**Comment by PlantDr430** [comment]
I started having a problem with Flash on one of my runs. I have already successfully completed 6 assemblies today using the same pipeline, however, this particular pair of reads is giving me trouble. I keep making it to Flash and get the 256 running command error a little after flash starts. I've tried increasing the ulimit, which allowed Flash to run slightly longer with this set, but ulimit is set to 8,192 and I can't increase it any higher. 

The Flash error is below and is in the log file attached. I'm just curious as to why this particular set of reads is failing when I have already done multiple runs today.

"Error writing "./flash.extendedFrags.fastq.gz"[FLASH] 
FLASH did not complete successfully; exiting with failure status (1)


 Here is my run: 
Homepage is https://github.com/tseemann/shovill
Operating system is linux
Using seqtk - /usr/bin/seqtk
Using pigz - /usr/bin/pigz
Using kmc - /usr/bin/kmc
Using trimmomatic - /usr/bin/trimmomatic
Using lighter - /usr/bin/lighter
Using flash - /usr/bin/flash
Using spades.py - /usr/bin/spades.py
Using bwa - /usr/bin/bwa
Using samtools - /usr/local/bin/samtools
Using java - /usr/bin/java
Using pilon - /usr/bin/pilon
Using tee - /usr/bin/tee
Using vmstat - /usr/bin/vmstat
Machine has 24 CPU cores and 251.900695800781 GB RAM
Changing into folder: /home/stephenwyka/ClavicepsGenomes/MiaoGenomes/LM71_shovill_output
Collecting raw read statistics with 'seqtk'
Running: seqtk fqchk -q3 \/home\/stephenwyka\/ClavicepsGenomes\/MiaoGenomes\/LM71J_2_S2_R1_001\.fastq\.gz > 10-seqtk.tab
Read stats: min_len = 35
Read stats: total_bp = 6272613692
Read stats: avg_len = 148
Read stats: max_len = 151
Estimating genome size with 'kmc'
Running: kmc -m8 -sm -n256 -ci3 -k25 -t24 \/home\/stephenwyka\/ClavicepsGenomes\/MiaoGenomes\/LM71J_2_S2_R1_001\.fastq\.gz kmc /tmp/0QLBjT60lN >> 20-kmc.log 2>&1
Using genome size 34235164 bp
Estimated sequencing depth: 183 x
Subsampling reads by factor 0.519 to get from 183x to 95x
Running: seqtk sample \/home\/stephenwyka\/ClavicepsGenomes\/MiaoGenomes\/LM71J_2_S2_R1_001\.fastq\.gz 0.519 | pigz --fast -c -p 24 > R1.sub.fq.gz
Running: seqtk sample \/home\/stephenwyka\/ClavicepsGenomes\/MiaoGenomes\/LM71J_2_S2_R2_001\.fastq\.gz 0.519 | pigz --fast -c -p 24 > R2.sub.fq.gz
Running: ln -s R1\.sub\.fq\.gz R1.fq.gz
Running: ln -s R2\.sub\.fq\.gz R2.fq.gz
Average read length looks like 148 bp
Setting k-mer range to (31 .. 111)
Estimated K-mers: 31 51 71 91 111 [kn=5, ks=20, kmin=31, kmax=111]
Using kmers: 31,51,71,91,111
Correcting reads with 'Lighter'
Running: lighter -od . -r R1.fq.gz -r R2.fq.gz -K 32 34235164 -t 24 -maxcor 1 >> 40-lighter.log 2>&1
Overlapping reads with 'FLASH'
Running: flash -m 20 -M 151 -d . -o flash -z -t 24 R1.cor.fq.gz R2.cor.fq.gz >> 50-flash.log 2>&1
Error 256 running command

And the three available log files are here
[20-kmc.log](https://github.com/tseemann/shovill/files/2061031/20-kmc.log)
[40-lighter.log](https://github.com/tseemann/shovill/files/2061032/40-lighter.log)
[50-flash.log](https://github.com/tseemann/shovill/files/2061033/50-flash.log)

Thank you

**Comment by PlantDr430** [comment]
Nevermind, problem was resolved.

**Comment by tseemann** [comment]
@PlantDr430 thats good to hear!  I have made the `shovill.log` MUCH better now.

**Comment by bifidoman** [comment]
> Below you can see the error message in shovel/20-kmc.log file:
> 
> Error: Cannot open temporary file /var/folders/mp/rjltg4g177v1jqwywpw2gjn00000gn/T/PJj2WG5Olm/kmc_00253.bin
> 

Hi,

just FYI
I got the same error, and I noticed that the tmp folder did not exist. I created the folder and It worked.
Shovill 0.9.0 on Mac Catalina
 best

marc

**Comment by tseemann** [comment]
@bifidoman lots of bug fixes since 0.9.0 - try and upgrade to 1.0.8 if you can!

**Comment by emmannaemeka** [comment]
I am having smilier issue 

Using seqtk - /Users/emmannaemeka/miniconda3/bin/seqtk
Using pigz - /Users/emmannaemeka/miniconda3/bin/pigz
Using kmc - /Users/emmannaemeka/miniconda3/bin/kmc
Using trimmomatic - /Users/emmannaemeka/miniconda3/bin/trimmomatic
Using lighter - /Users/emmannaemeka/miniconda3/bin/lighter
Using flash - /Users/emmannaemeka/miniconda3/bin/flash
Using spades.py - /Users/emmannaemeka/miniconda3/bin/spades.py
Using bwa - /Users/emmannaemeka/miniconda3/bin/bwa
Using samtools - /usr/local/bin/samtools
Using java - /Users/emmannaemeka/miniconda3/bin/java
Using pilon - /Users/emmannaemeka/miniconda3/bin/pilon
Using tee - /usr/bin/tee
Using sysctl - /usr/sbin/sysctl
Machine has 4 CPU cores and 16 GB RAM
Forced removal of existing --outdir spo5_shovill (please wait)
Changing into folder: /Users/emmannaemeka/Documents/Research/salmonella_phage_project/spo5/spo5_shovill
Collecting raw read statistics with 'seqtk'
Running: seqtk fqchk -q3 \/Users\/emmannaemeka\/Documents\/Research\/salmonella_phage_project\/spo5\/21\-005\-0894_R1\.forward_paired\.fq\.gz > 10-seqtk.tab
Read stats: min_len = 36
Read stats: max_len = 151
Read
[20-kmc.log](https://github.com/tseemann/shovill/files/7692484/20-kmc.log)
 stats: total_bp = 385940918
Read stats: avg_len = 148
Estimating genome size with 'kmc'
Running: kmc -m8 -sm -n256 -ci3 -k25 -t4 \/Users\/emmannaemeka\/Documents\/Research\/salmonella_phage_project\/spo5\/21\-005\-0894_R1\.forward_paired\.fq\.gz kmc /Users/emmannaemeka/Documents/Research/salmonella_phage_project/spo5/W27fF0qrUD >> 20-kmc.log 2>&1
Error 256 running command

I tried to change temporary folder 

shovill --outdir  spo5_shovill --R1 21-005-0894_R1.forward_paired.fq.gz  --R2 21-005-0894_R2.reverse_paired.fq.gz --tmpdir /Users/emmannaemeka/Documents/Research/salmonella_phage_project/spo5


When I ran less /var/folders/8v/08h9bb_s5jl6_f70nwzmj6jr0000gn/T/WSPCtMFcxb/kmc_00253.bin
/var/folders/8v/08h9bb_s5jl6_f70nwzmj6jr0000gn/T/WSPCtMFcxb/kmc_00253.bin: No such file or directory

How can I solve this?
